### PR TITLE
CR-7598

### PR DESCRIPTION
**Pull request by elad-codefresh**

### CODE REVIEW

File: pkg/store/store.go [hunk 1]
```diff
@@ -183,6 +186,8 @@ func init() {
 	s.GithubAccessTokenSecretObjectName = "autopilot-secret"
 	s.GithubAccessTokenSecretKey = "git_token"
 	s.ArgoCD = "argo-cd"
+	s.MinimumMemorySizeRequired = "3000"
```
Comment (elad-codefresh) [inline pkg/store/store.go]: what should we set as the minimum requirements?

File: cmd/commands/runtime.go [hunk 2]
```diff
@@ -292,6 +307,172 @@ func createRuntimeOnPlatform(ctx context.Context, opts *model.RuntimeInstallatio
...
+	
+	client, err := kubeFactory.KubernetesClientSet()
+	if err != nil {
+		return fmt.Errorf(fmt.Sprintf("Cannot create kubernetes clientset: %v ", err))
+	}
+
+	req := validationRequest{
+		rbac: []rbacValidation{
+			{
+				Resource:  "ServiceAccount",
```
Comment (elad-codefresh) [inline cmd/commands/runtime.go]: wondering if in this case I should change the hard-coded verb values and resource names to constants defined in the store

File: cmd/commands/runtime.go [hunk 3]
```diff
@@ -292,6 +307,172 @@ func createRuntimeOnPlatform(ctx context.Context, opts *model.RuntimeInstallatio
 	return runtimeCreationResponse.NewAccessToken, nil
 }
 
+func ensureClusterRequirements(ctx context.Context, kubeFactory kube.Factory, namespace string) error {
+	requirementsValidationErrorMessage := "cluster does not meet minimum requirements"
+	var specificErrorMessages []string
+	
+	client, err := kubeFactory.KubernetesClientSet()
+	if err != nil {
+		return fmt.Errorf("%s: cannot create kubernetes clientset: %v ", requirementsValidationErrorMessage, err)
```
Comment (roi-codefresh) [inline cmd/commands/runtime.go]: no need for the `cluster does not meet minimum requirements` here. maybe just `failed to verify cluster requirements: cannot create kubernetes clientset: %w`...

File: cmd/commands/runtime.go (hunk 2 above)
Comment (roi-codefresh) [reply cmd/commands/runtime.go]: no need

File: cmd/commands/runtime.go [hunk 4]
```diff
@@ -292,6 +307,172 @@ func createRuntimeOnPlatform(ctx context.Context, opts *model.RuntimeInstallatio
...
+				Namespace: namespace,
+			},
+			{
+				Resource:  "Service",
+				Verbs:     []string{"create", "update", "delete"},
+				Namespace: namespace,
+			},
+			{
+				Resource:  "Role",
+				Group:     store.Get().RoleApiGroup,
```
Comment (roi-codefresh) [inline cmd/commands/runtime.go]: no need to save it to the store

File: cmd/commands/runtime.go [hunk 5]
```diff
@@ -292,6 +307,172 @@ func createRuntimeOnPlatform(ctx context.Context, opts *model.RuntimeInstallatio
...
+		}
+		return fmt.Errorf("%s: failed testing rbac: %v", requirementsValidationErrorMessage, specificErrorMessages)
+	}
+
+	nodes, err := client.CoreV1().Nodes().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return fmt.Errorf("%s: failed getting nodes: %v", requirementsValidationErrorMessage, err)
+	}
+
+	if nodes == nil {
```
Comment (roi-codefresh) [inline cmd/commands/runtime.go]: don't think there's a need for this check

File: cmd/commands/runtime.go [hunk 6]
```diff
@@ -292,6 +307,172 @@ func createRuntimeOnPlatform(ctx context.Context, opts *model.RuntimeInstallatio
 	return runtimeCreationResponse.NewAccessToken, nil
 }
 
+func ensureClusterRequirements(ctx context.Context, kubeFactory kube.Factory, namespace string) error {
```
Comment (roi-codefresh) [inline cmd/commands/runtime.go]: please move this function and everything related to it to pkg/util/kube/kube.go

File: cmd/commands/runtime.go [hunk 7]
```diff
@@ -47,6 +47,7 @@ import (
 	argowf "github.com/argoproj/argo-workflows/v3/pkg/apis/workflow"
 
 	"github.com/Masterminds/semver/v3"
+	cliv2kube "github.com/codefresh-io/cli-v2/pkg/util/kube"
```
Comment (elad-codefresh) [inline cmd/commands/runtime.go]: tested locally with 'replace'

File: pkg/store/store.go [hunk 8]
```diff
@@ -183,6 +186,8 @@ func init() {
...
```
Comment (elad-codefresh) [inline pkg/store/store.go:189]: are we ok with these minimum requirements?

File: cmd/commands/runtime.go (hunk 7 above)
Comment (roi-codefresh) [reply cmd/commands/runtime.go]: name it as `kubeutil` i think it's a better name

**Review by roi-codefresh — APPROVED** [review]

**Pull request merged**